python/cudf/cudf/core/column/timedelta.py

```diff
@@ -273,9 +260,9 @@ def as_numerical(self) -> "cudf.core.column.NumericalColumn":
             ),
         )
 
-    @property
+    @functools.cached_property
     def time_unit(self) -> str:
-        return self._time_unit
+        return np.datetime_data(self.dtype)[0]
```

Comment:
**reminder** If we ever support non-unit datetimes (e.g. `datetime64[25s]`) this code will need to change.